pkg/minishift/addon/command/interpolation_context.go

```diff
@@ -70,13 +70,13 @@ func (c *defaultInterpolationContext) RemoveFromContext(key string) error {
 
 func (c *defaultInterpolationContext) Interpolate(cmd string) string {
 	for _, v := range c.context {
-		cmd = v.regexp.ReplaceAllLiteralString(cmd, v.subsitution)
+		cmd = v.regexp.ReplaceAllLiteralString(cmd, v.substitution)
 	}
 	return cmd
 }
 
 func (c *defaultInterpolationContext) Vars() []string {
-	keys := make([]string, len(c.context))
+	keys := make([]string, 0, len(c.context))
```

Comment:
Why starting with a 0 size if you know exactly how many elements you are going to add?

Reply:
Seems best practice to minimize memory allocations https://stackoverflow.com/a/21363587/1120530.

Reply:
@hferentschik , looks like if we add length 2, we will not get the expected result https://play.golang.org/p/wFEQUJYUKb

Reply:
We could have achieve, same with `keys := make([]string, 0)` as well. Seems that Golang slice will grow organically as the elements get appended. But to minimize memory allocations, `keys := make([]string, 0, 2)` is the suggested approach.